stratum/hal/lib/tdi/tdi_extern_manager.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef STRATUM_HAL_LIB_TDI_TDI_EXTERN_MANAGER_H_
+#define STRATUM_HAL_LIB_TDI_TDI_EXTERN_MANAGER_H_
+
+#include "absl/memory/memory.h"
+#include "idpf/p4info.pb.h"
```

Comment:
To clarify, we dont use IDPF in other targets except ES2K right?
This introduces it at TDI layer -- which then inherits to DPDK and Tofino too, doesnt it? Is it possible to move it further down to ES2K only?

Reply:
I didn't introduce the dependency; it was already there. I filed a separate issue for this (https://github.com/ipdk-io/stratum-dev/issues/290).

TdiTableManager #includes `idp4/p4info.pb.h` so it can support PacketModMeter and DirectPacketModMeter.

My original PR addressed this issue as well. This involved making a considerable number of changes. I realized this was too much to do at one time, so I stripped it out. This is what I meant when I talked about "dialing it back."

My original implementation of TdiExternManager omitted the Find methods for the PacketMod meters. This meant that I had to downcast the TdiExternManager pointer to an Es2kExternManager pointer in order to call the methods. I decided it would be less ugly if I added the Find methods to TdiExternManager as well. I moved the aberrant #include from tdi_table_manager.cc to tdi_extern_manager.h, effectively consolidating the ugliness in one place. Not ideal, but an improvement over the original code.

The most pressing problem right now is making P4InfoManager upstreamable. Once that's done, we can tackle the TDI dependency as a separate problem.